library/include/rocwmma/internal/blend_impl.hpp

```diff
@@ -105,8 +107,12 @@ namespace rocwmma
                 template <typename DataT>
                 ROCWMMA_DEVICE static inline DataT exec(DataT src0, DataT src1)
                 {
+                    static_assert(sizeof(DataT) == sizeof(uint32_t), "Inputs must be 32 bit");
                     uint32_t const mask = MaskCtrl::maskCtrl();
-                    return (src1 & mask) | (src0 & ~mask);
+                    reinterpret_cast<uint32_t&>(src0)
```

Comment:
What is the need for these reinterpret_casts after the assertion of 32bit size?

Reply:
Bitwise ops require integer types, this one we are expecting b32 size